### PR TITLE
WT-10382 Update assertion message in trigger_abort

### PR DESCRIPTION
**Pull request by clarissecheah**
Currently the `trigger_abort` function has a generic failure message. This PR updates the message to reflect a WiredTiger failure and prints the log output in a single line instead of separate lines. This is done to support the creation of an Atlas [log parsing rule](https://jira.mongodb.org/browse/CLOUDP-159804) for [PM-2897](https://jira.mongodb.org/browse/PM-2897)

### CODE REVIEW

File: src/include/error.h [hunk 1]
```diff
@@ -143,32 +159,27 @@
...
     } while (0)
 #else
-#define TRIGGER_ABORT(session, exp, ...)                     \
-    do {                                                     \
-        __wt_errx(session, "Assertion '%s' failed: ", #exp); \
-        __wt_errx(session, __VA_ARGS__);                     \
-        __wt_abort(session);                                 \
+#define TRIGGER_ABORT(session, exp, ...)                                             \
+    do {                                                                             \
+        char _buf[WT_ERR_MSG_BUF_LEN];                                               \
```
Comment (jiechenbo) [inline src/include/error.h:179]: Do we need the unittest global buffer if we have one localised now?

Comment (clarissecheah) [reply src/include/error.h:175]: There are two different definitions of `trigger_abort`, one for when we're unit testing and the other when we're not. The buffer you're referring to (`unittest_assert_msg`) is used specifically when unit testing to identify when the assertion has "fired".

File: src/include/error.h [hunk 2]
```diff
@@ -143,32 +159,27 @@
...
-        (session)->unittest_assert_hit = true;                                                  \
+#define TRIGGER_ABORT(session, exp, ...)                                                    \
+    do {                                                                                    \
+        if ((session) == NULL) {                                                            \
+            __wt_errx(                                                                      \
+              session, "A non-NULL session must be provided when unit testing assertions"); \
+            __wt_abort(session);                                                            \
+        }                                                                                   \
+        /*                                                                                  \
+         * For Atlas log parsing purposes.                                                  \
```
[the remark this answers is not in the document]
Comment (clarissecheah) [reply src/include/error.h]: I'll remove the Atlas references altogether as it might just cause more confusion than explanation.

**Review by jiechenbo — APPROVED** [review]
LGTM!

**Pull request merged**